test/v1_2_0/integration/withdrawals/Withdrawals.sol

```diff
@@ -0,0 +1,136 @@
+pragma solidity ^0.8.17;
+
+import {EscrowBase, IAddressGaugeVote} from "../../base/EscrowBase.sol";
+import {console2 as console} from "forge-std/console2.sol";
+import {Vm} from "forge-std/Vm.sol";
+import {IERC721Receiver} from "@openzeppelin/contracts/token/ERC721/IERC721Receiver.sol";
+
+import {
+    Clock,
+    IClock,
+    Lock,
+    VotingEscrow,
+    IVotingEscrowIncreasing,
+    IEscrowCurveIncreasing,
+    IVotingEscrowIncreasing,
+    IVotingEscrowCoreErrors,
+    IMerge,
+    ISplit,
+    ILockedBalanceIncreasing,
+    IEscrowCurveGlobalStorage,
+    IEscrowCurveTokenStorage,
+    IEscrowCurveGlobalStorage,
+    EscrowIVotesAdapter,
+    IEscrowIVotesAdapterErrorsAndEvents,
+    IGaugeVote
+} from "../../versions.sol";
+
+contract ERC721ReceiverMock is IERC721Receiver {
+    function onERC721Received(
+        address,
+        address,
+        uint256,
+        bytes calldata
+    ) external pure override returns (bytes4) {
+        return this.onERC721Received.selector;
+    }
+}
+
+contract TestWithdrawal is IEscrowCurveTokenStorage, IEscrowCurveGlobalStorage, EscrowBase {
+    address gauge = address(0x777);
+    address alice = address(0x123);
+    address bob = address(0x192);
+
+    function setUp() public override {
+        super.setUp();
+
+        vm.warp(2 weeks + 1 hours + 1);
+        voter.createGauge(gauge, "metadata");
+        escrow.enableSplit();
+    }
+
+    struct User {
+        address user;
+        uint96 amount;
+        bool withdraws;
+    }
+
+    function testFuzz_WithrawWithCancel(User[20] memory _users) public {
+        uint256[] memory tokenIds = new uint256[](_users.length);
+        IAddressGaugeVote.GaugeVote[] memory votes = new IAddressGaugeVote.GaugeVote[](1);
+        votes[0] = IAddressGaugeVote.GaugeVote(100, gauge);
+
+        // Assume no duplicate addresses are found.
```

Comment:
easier to just populate an array w. 20 random addresses derived from hash of a seed + i